WordPress/openverse-frontend
# Fix search bar button crashing index page

## Fixes


Fixes #[issue number] by @[issue author]

## Description


The root cause of this was that the contents of the VButton and VIconButton were not both another element. The VButton was rendering a plain string and VIconButton rendered an SVG. When Vue tried to evaluate the tag name of the VButton when it was just the text, it would fail because it was just a text node which has no tag name.

This _could_ just be a one-line fix by wrapping the contents of the VButton in a `span` BUT I think the VSearchButton component is actually simpler this way, uses less markup and less CSS _and_ works better with SSR (no more flashing "Search" text or icon for most cases of mobile/desktop usage). The only people who still get a little flashing of the search/icon are users of non-mobile devices with narrow viewports.

Something is currently weird about the search button. I don't know why it doesn't have a pink background.

## Testing Instructions

Load the homepage on a mobile device browser and verify it doesn't crash. Also, verify going back to the homepage from the search page.

Also e2e and unit testss

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
